src/widgets/table/table.rs

```diff
@@ -223,16 +223,16 @@ pub struct Table<'a> {
 impl<'a> Default for Table<'a> {
     fn default() -> Self {
         Self {
-            rows: Default::default(),
-            header: Default::default(),
-            footer: Default::default(),
-            widths: Default::default(),
+            rows: Vec::new(),
+            header: None,
+            footer: None,
+            widths: Vec::new(),
             column_spacing: 1,
-            block: Default::default(),
-            style: Default::default(),
-            highlight_style: Default::default(),
-            highlight_symbol: Default::default(),
-            highlight_spacing: Default::default(),
+            block: None,
+            style: Style::new(),
+            highlight_style: Style::new(),
```

Comment:
Default can not be const so I personally would prefer using const new when possible. That should allow for more compiler optimisations.

(no it shouldn’t trigger a lint)

There are more cases where new over default would allow for more const optimisations but I don’t think that’s part of PR as it includes more refactoring.

Reply:
This is not a blocking comment - so can be left alone or fixed at your preference. marking it as resolved for now.